### PR TITLE
Refactor ConfigValidator

### PR DESCRIPTION
**Pull request by bogdandrutu**
Followup after #2802

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: config/receiver.go [hunk 1]
```diff
@@ -16,27 +16,17 @@ package config
 
 // Receiver is the configuration of a receiver. Specific receivers must implement this
 // interface and will typically embed ReceiverSettings struct or a struct that extends it.
-// Embedded CustomConfigOptions will force each receiver to implement Validate() function
+// Embedded CustomValidator will force each receiver to implement Validate() function
 type Receiver interface {
 	NamedEntity
-	CustomConfigOptions
+	CustomValidator
```
Comment (bogdandrutu) [inline config/receiver.go:22]: Maybe just `Validator`?

File: config/config.go [hunk 2]
```diff
@@ -160,6 +160,12 @@ type NamedEntity interface {
 	SetName(name string)
 }
 
+// Validator defines the interface for the configuration validation.
+type Validator interface {
```
Comment (tigrannajaryan) [inline config/config.go:164]:
```suggestion
type Validatable interface {
```

File: config/config.go [hunk 3]
```diff
@@ -108,13 +115,6 @@ func (cfg *Config) validateServicePipelines() error {
 			if cfg.Receivers[ref] == nil {
 				return fmt.Errorf("pipeline %q references receiver %q which does not exist", pipeline.Name, ref)
 			}
-
-			// Validate the receiver configuration by the custom configuration validator
-			recCfg := cfg.Receivers[ref]
-			if err := recCfg.Validate(); err != nil {
-				return fmt.Errorf("pipeline %q references receiver %q which has invalid configuration with error: %v", pipeline.Name, ref, err)
-			}
```
Comment (tigrannajaryan) [inline config/config.go:116]: Do you know why this was not in `(cfg *Config) Validate()` earlier? Is it possible this is somehow important?

**Review by tigrannajaryan — APPROVED** [review]
Generally LGTM with minor comments.

File: config/config.go (hunk 2 above)
Comment (bogdandrutu) [reply config/config.go]: Done, also for the moment make it unexported

File: config/config.go (hunk 3 above)
Comment (bogdandrutu) [reply config/config.go:116]: No specific reason, it was added in the previous commit.

**Pull request merged**